src/proc/decimation-filter.cpp

```diff
@@ -163,21 +169,45 @@ namespace librealsense
                     p = pixel_raws[n] + chunk_offset;
                     for (size_t m = 0; m < scale; ++m)
                     {
-                        *wk_itr++ = *(p + m);
-                        //*wk_itr++ = *(pixel_raws[n] + chunk_offset + m);
-                        //working_kernel[n*scale + m] = *(pixel_raws[n] + chunk_offset + m);
+                        if (*(p + m))
+                            *wk_itr++ = *(p + m);
                     }
                 }
 
-                std::nth_element(wk_begin, wk_begin + (kernel_size / 2), wk_begin + kernel_size);
-                //std::sort(working_kernel.begin(),working_kernel.end());
-                *frame_data_out++ = working_kernel[kernel_size / 2];
+                // For even-size kernels pick the member one below the middle
+                auto ks = (int)(wk_itr - wk_begin);
+                if (ks == 0)
+                    *frame_data_out++ = 0;
+                else
+                {
+                    //// Mean value shall be used by reference design if elements are more than 4.
```

Comment:
I only see the implementation for median, not mean

Reply:
It is explained in the next line (multi-line comment). I'll refactor it to be more concise